src/query/v2/context.hpp

```diff
@@ -42,21 +43,27 @@ struct EvaluationContext {
   mutable std::unordered_map<std::string, int64_t> counters;
 };
 
-inline std::vector<storage::v3::PropertyId> NamesToProperties(const std::vector<std::string> &property_names,
-                                                              DbAccessor *dba) {
+inline std::vector<storage::v3::PropertyId> NamesToProperties(
+    const std::vector<std::string> &property_names, msgs::ShardRequestManagerInterface *shard_request_manager) {
   std::vector<storage::v3::PropertyId> properties;
   properties.reserve(property_names.size());
   for (const auto &name : property_names) {
-    properties.push_back(dba->NameToProperty(name));
+    if (shard_request_manager != nullptr) {
+      properties.push_back(shard_request_manager->NameToProperty(name));
+    }
   }
   return properties;
 }
 
-inline std::vector<storage::v3::LabelId> NamesToLabels(const std::vector<std::string> &label_names, DbAccessor *dba) {
+inline std::vector<storage::v3::LabelId> NamesToLabels(const std::vector<std::string> &label_names,
+                                                       msgs::ShardRequestManagerInterface *shard_request_manager) {
   std::vector<storage::v3::LabelId> labels;
   labels.reserve(label_names.size());
   for (const auto &name : label_names) {
-    labels.push_back(dba->NameToLabel(name));
+    // TODO Fix by using reference
```

Comment:
I think this TODO doesn't make too much sense here, even though I might be the author of it.

Reply:
I think I meant it for `shard_request_manager`, so it is never nullptr